strongloop/loopback-next
# docs: migrate loopback-example-passport repo as lb4 example

satisfies: https://github.com/strongloop/loopback-next/issues/3958


## Test steps

** This example repo would need manual test

* preconditions: create a test app and test user in facebook, and make sure to give `email` permission to test user. (emonddr : visit https://developers.facebook.com to do this.) 
* preconditions: Copy test app's client id and secret and update those in the examples/passport-poauth2-login/authentication-strategies/oauth2-strategy.ts

* Run `npm i` from loopback-next root folder, to create symlinks with master packages
* cd examples/passport-oauth2-login & npm start
* open browser -> http://localhost:3000  (emonddr : Open a new window in Incognito mode, otherwise may not work)
* click login -> login with facebook 
* page redirects to facebook, enter test user credentials
* if authentication was successful, page redirects to login profiles page and shows linked facebook account details along with access token

## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-next) 👈

- [x] `npm test` passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] API Documentation in code was updated
- [ ] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈
